src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -673,6 +673,54 @@ public void BuildWithTransitiveReferenceToNetCoreAppPackage()
                 .Pass();
         }
 
+        [WindowsOnlyFact]
+        public void It_escapes_resolved_package_assets_paths()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "ProjectWithPackageThatNeedsEscapes",
+                TargetFrameworks = "net462",
+                IsSdkProject = true,
+                IsExe = true,
+            };
+
+            testProject.SourceFiles["ExampleReader.cs"] = @"
+using System;
+using System.Threading.Tasks;
+
+namespace ContentFilesExample
+{
+    internal static class ExampleInternals
+    {
+        internal static Task<string> GetFileText(string fileName)
+        {
+            throw new NotImplementedException();
+        }
+    }
+}
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine(""Hello World!"");
+    }
+}";
+
+            testProject.PackageReferences.Add(new TestPackageReference("ContentFilesExample", "1.0.2"));
```

Comment:
Add an explanation of why we're referencing this package.  As I understand it, it's because it will resolve assets in a PCL directory in the package with `+` in the path, which needs to be escaped.

Reply:
(The package doesn't have a +, it has a %2B.)

Reply:
Which is super weird. Was this intentional? Why does nuget even match PLIB assets to portable-x%2By? 'm guessing there's a bug farm in msbuild for paths like that based on what ItemSpec getter does.

Do we have any other reports of real packages that are like this?

Reply:
Yeah, I'd really like to know if this is a NuGet issue (unescaping the file on disk for internal consumption), or a case where MSBuild overzealously escapes/unescapes.

Reply:
I think it's both. NuGet unescaped to find the asset was applicable, and MSBuild unescaped the path too. I think neither of these things should be unescaping.

Reply:
I don't think this package should have even gotten this far, but I'd love to understand if the %2Bs inside are meant to demonstrate a real scenario, and what that scenario is. @nkolev92

Reply:
While this particular package might have been erroneous, several of the MSBuild special characters are legal path characters on supported systems.

I don't have a particularly strong opinion on fixing this since this bug has been around for a very long time now and I'm guessing it has impacted very few, if any, users.  My argument for fixing it would simply be that the fix is simple, low-risk, and it makes asset resolution more robust to any paths contained in a package.

Reply:
I agree we should fix this in case someone has a folder with `%20` in it (or whatever). But I'd also like to understand the NuGet behavior.

Reply:
To @dsplaisted's original point, a comment here would be appreciated as to why this particular package is being used for the test.  I'll at least add that while we deliberate on whether or not this should be fixed.

Reply:
I'm OK with the fix (after it is moved to cache write). I'm also just trying to understand if nuget picking assets based on unescaped folder name is deliberate or a weird quirk that this package happens to rely on.

Reply:
Looking into it. 
There's a historical reason for the normalization since VS 2013 didn't work with non-normalized paths. 

NuGet pack used to escape paths, now it doesn't. https://github.com/NuGet/NuGet.Client/pull/1855

@nguerrera's comment is suggesting I should double check why/whether lib & contentFiles are treated differently anywhere in NuGet. 

I'll get back once I've looked into that.